lib/src/components/progress/progress_circle.dart

```diff
@@ -48,14 +52,42 @@ class ZetaProgressCircle extends ZetaProgress {
     properties
       ..add(EnumProperty<ZetaCircleSizes>('size', size))
       ..add(DoubleProperty('progress', progress))
-      ..add(DiagnosticsProperty<bool>('rounded', rounded));
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(ObjectFlagProperty<VoidCallback?>.has('onCancel', onCancel));
   }
 }
 
 ///Class definition for [ZetaProgressCircleState]
 class ZetaProgressCircleState extends ZetaProgressState<ZetaProgressCircle> {
+  final _controller = WidgetStatesController();
+
+  @override
+  void initState() {
```

Comment:
I don't think you need the initState or the listener, the state re-render is done already  by the `ListenableBuilder`